src/app/epics/user-login.epic.spec.ts

```diff
@@ -0,0 +1,176 @@
+import { UserLoginEpic } from './user-login.epic';
+import { ofType, ActionsObservable } from 'redux-observable';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/observable/of';
+import 'rxjs/add/observable/throw';
+
+import 'rxjs/add/observable/empty';
+import 'rxjs/add/operator/switchMap';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/catch';
+
+import { AnyAction } from 'redux';
+import { IUser } from '../models/user.model';
+import { UserActions } from '../actions/user.actions';
+
+let mockUser: IUser = { name: 'MOCK_user' };
+
+class MockLoginService {
+    login = (username, pass) => {};
+
+    logout = () => {};
+};
+// TODO: Don't feel confident in the testing of Observables.
+// TODO: Needs to be updated since Login Service has changed its error handle.
+//      And because there is now a Router dependency
+class MockUserActions {
+    loginSuccess = (user, date) => ({ type: 'MOCK_LOGIN_SUCCESS', user, date });
+    loginFailed = (err) => ({ type: 'MOCK_LOGIN_ERROR', err });
+};
+
+class MockRouter {
+    router = {
+        navigate: (commands, extras) => {},
+    };
+}
```

Comment:
The mocks themselves look fine - lately I've become a fan of doing `mockRouter = jasmine.createSpyObj('mockRouter', ['navigate']);` just a concise convenience tool - methods created through this are stubbed out for you